packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskScheduler.kt

```diff
@@ -0,0 +1,231 @@
+        Log.e(TAG, "Worker failed to start with error " + e.message)
+        false
+      }
+    }
+
+    /**
+     * Cancels the worker task
+     */
+    suspend fun stopWorker(context: Context): Boolean {
+      if (!isWorkerRunning(context)) {
+        return false
+      }
+
+      Log.d(TAG, "Cancelling worker with identifier $WORKER_IDENTIFIER")
+
+      // Stop our main worker
+      val workManager = WorkManager.getInstance(context)
+      return try {
+        workManager.cancelUniqueWork(WORKER_IDENTIFIER).await()
+      } catch (e: Exception) {
+        Log.d(TAG, "Stopping worker failed with error " + e.message)
+        false
+      }
+    }
+
+    /**
+     * Returns true if the worker task is pending
+     */
+    private suspend fun isWorkerRunning(context: Context): Boolean {
+      val workInfo = getWorkerInfo(context)
+      return workInfo?.state == WorkInfo.State.RUNNING ||
+        workInfo?.state == WorkInfo.State.ENQUEUED
+    }
+
+    /**
+     * Runs tasks with the given appScopeKey
+     */
+    suspend fun runTasks(context: Context, appScopeKey: String) : Boolean {
+      // Get task service
+      val taskService = TaskServiceProviderHelper.getTaskServiceImpl(context)
+        ?: throw MissingTaskServiceException()
+
+      Log.d(TAG, "runTasks: $appScopeKey")
+
+      // Get all task consumers
+      val consumers = taskService.getTaskConsumers(appScopeKey)
+      Log.d(TAG, "runTasks: number of consumers ${consumers.size}")
+
+      if (consumers.size == 0) {
+        return false
+      }
+
+      val tasks = consumers.mapNotNull { consumer ->
+        val bgTaskConsumer = consumer as? BackgroundTaskConsumer ?: return@mapNotNull null
+
+        Log.d(TAG, "runTasks: executing tasks for consumer of type ${consumer.taskType()}")
+
+        val taskCompletion = CompletableDeferred<Unit>()
+
+        bgTaskConsumer.executeTask {
+          Log.d(TAG, "Task successfully finished")
+          taskCompletion.complete(Unit)
+        }
+
+        taskCompletion
+      }
+
+      // Await all tasks to complete
+      tasks.awaitAll()
+
+      return true
+    }
+
+    /**
+     * Returns the worker info object from the WorkManager if the worker has been
+     * registered, otherwise returns null
+     */
+    private suspend fun getWorkerInfo(context: Context): WorkInfo? {
+      // Get work manager
+      val workManager = WorkManager.getInstance(context)
+
+      return try {
+        val workInfos = workManager.getWorkInfosForUniqueWork(WORKER_IDENTIFIER).await()
+        return workInfos.firstOrNull()
+      } catch (e: Exception) {
+        Log.d(TAG, "Calling getWorkInfosForUniqueWork failed with error " + e.message)
+        null
+      }
+    }
+
+    /**
+     * Helper function for calling functions returning an Operation
+     */
+    suspend fun Operation.await(): Boolean = withContext(Dispatchers.IO) {
+      try {
+        result.get()
+        true
+      } catch (e: Exception) {
+        throw e
+      }
```

Comment:
```suggestion
      result.get()
      true
```
This try-catch seems to be unnecessary 🤔